common/physics-system.cpp

```diff
@@ -93,21 +102,49 @@ namespace tec {
 				body->clearForces();
 			}
 
+			// handle changes to desired deactivation mode
 			if (collidable->disable_deactivation) {
 				body->forceActivationState(DISABLE_DEACTIVATION);
 			}
 			else if (body->getActivationState() == DISABLE_DEACTIVATION) {
 				body->forceActivationState(ACTIVE_TAG);
 			}
 
+			// prevent the simulation from rotating the object
+			// this doesn't account for change after creation, once disabled there isn't a re-enable
 			if (collidable->disable_rotation) {
```

Comment:
You must remove the rigidbody IIRC to make any of these flag updates take affect.